sklearn/metrics/_regression.py

```diff
@@ -1197,3 +1201,235 @@ def d2_tweedie_score(y_true, y_pred, *, sample_weight=None, power=0):
     denominator = np.average(dev, weights=sample_weight)
 
     return 1 - numerator / denominator
+
+
+def d2_pinball_loss_score(
+    y_true, y_pred, *, sample_weight=None, alpha=0.5, multioutput="uniform_average"
+):
+    """
+    :math:`D^2` regression score function, \
+    fraction of pinball loss deviance explained.
+
+    Best possible score is 1.0 and it can be negative (because the model can be
+    arbitrarily worse). A model that always uses the empirical median of `y_true`
+    as constant prediction, disregarding the input features,
+    gets a :math:`D^2` score of 0.0.
+
+    Read more in the :ref:`User Guide <d2_score>`.
+
+    .. versionadded:: 1.1
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,) or (n_samples, n_outputs)
+        Ground truth (correct) target values.
+
+    y_pred : array-like of shape (n_samples,) or (n_samples, n_outputs)
+        Estimated target values.
+
+    sample_weight : array-like of shape (n_samples,), optional
+        Sample weights.
+
+    alpha : float, default=0.5
+        Pinball loss quantile parameter, determines the slope of the pinball_loss.
+        Equivalent to `d2_absolute_error_score` when `alpha=0.5`.
+
+    multioutput : {'raw_values', 'uniform_average'} or array-like of shape \
+            (n_outputs,), default='uniform_average'
+        Defines aggregating of multiple output values.
+        Array-like value defines weights used to average scores.
+
+        'raw_values' :
+            Returns a full set of errors in case of multioutput input.
+
+        'uniform_average' :
+            Scores of all outputs are averaged with uniform weight.
+
+    Returns
+    -------
+    score : float or ndarray of floats
+        The :math:`D^2` score with a pinball loss deviance
+        or ndarray of scores if `multioutput='raw_values'`.
+
+    Notes
+    -----
+    This is not a symmetric function.
+
```

Comment:
```suggestion
```

Reply:
I think that D2 is not a symmetric function because if you switch `y_true` with `y_pred` the denominator is no longer correct, right? Or am I missing something?

Reply:
Yes, but I think its not relevant enough to be mentioned. If, at all, it should be mentioned that squared error and absolute error are symmetric.

Reply:
Good point. I agree there's no reason to assume it's a symmetric function in the first place. I only added that to conform with the convention in [r2_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html#sklearn.metrics.r2_score) and [d2_tweedie_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.d2_tweedie_score.html#sklearn.metrics.d2_tweedie_score). Should I remove it from these two as well to be consistent?